### PR TITLE
Feature#16 add functionality to the loginview

### PR DESCRIPTION
**Pull request by EusebioAjas**
# Description
 - What?: We've added email and password authentication  in LoginView. Also We've added loading spinner. 
 - Why?: These changes complete the user login functionality.

  
  ## Screenshots 
LoginSpinner

### CODE REVIEW

File: src/screens/LoginView/LoginView.jsx [hunk 1]
```diff
@@ -6,43 +6,51 @@ import TextBox from '../../components/TextBox/TextBox';
...
-      </View>
-    </ScrollView>
+    <>
+      {loading && <Spinner text='loging in'/>}
+      <ScrollView style={styles.container}>
+        <Text style={styles.title}>Sign In</Text>
+        <View style={styles.inputContainer}>
+          <InputLabel errorMessages={getErrorsInField('email')}>Email</InputLabel>
+          <TextBox style={styles.textInputLabel} value={email} onChange={setEmail} />
+        </View>
```
Comment (DemisRincon) [inline src/screens/LoginView/LoginView.jsx:39]: this could be a component , codeclimate will detect reapedted code

**Review by DemisRincon — APPROVED** [review]

File: src/library/methods/firebaseLogin.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+      if (error.code === 'auth/invalid-email') {
+        alert('That email address is invalid!');
+      }
+
+      if(error.code == 'auth/invalid-password') {
+        alert('Invalid password');
+      }
+
+      alert('Couldn´t log in');
```
Comment (YilKanda) [inline src/library/methods/firebaseLogin.js:22]: ESLint indicates that there is a syntactic error when placing only alert, I recommend using [Alert.alert()](https://reactnative.dev/docs/alert)

File: src/components/Spinner/Spinner.jsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    <View style={styles.container}>
+      <View style={styles.spinnerContainer}>
+        <ActivityIndicator size="large" color={colors.blue}/>
+        <Text style={styles.text}>{text}</Text>
+      </View>
+    </View>
+  )
+};
+
+export default Spinner;
```
Comment (YilKanda) [inline src/components/Spinner/Spinner.jsx:17]: Add a blank line.

File: src/components/Spinner/SpinnerStyle.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    top: 0,
+    bottom: 0,
+    justifyContent: 'center',
+    alignItems: 'center',
+    zIndex: 10,
+    backgroundColor: 'rgba(0, 0, 0, 0.4)',
+  },
+  spinnerContainer: {
+    top: -10,
+    backgroundColor: '#21252b',
```
Comment (YilKanda) [inline src/components/Spinner/SpinnerStyle.js:18]: In this line, you are not using the option to call your color library.

File: src/screens/LoginView/LoginView.jsx [hunk 5]
```diff
@@ -6,43 +6,51 @@ import TextBox from '../../components/TextBox/TextBox';
...
-  const {validate, isFieldInError, getErrorsInField, getErrorMessages} = useValidation({state: { email, password }});
+  const {validate, isFieldInError, getErrorsInField} = useValidation({state: { email, password }});
 
-  const onPressButton = () => {
+  const onPressButton = async () => {
     validate({
       email: { email: true, required: true },
       password: { minlength: 8, required: true },
-    });
+    });
```
Comment (YilKanda) [inline src/screens/LoginView/LoginView.jsx:23]: There are unnecessary spaces here.

```suggestion
    });
```

**Review by jkzasori — APPROVED** [review]

**Pull request merged**